strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/jobs/PerRepositoryDuplicationStrategy.java

```diff
@@ -0,0 +1,81 @@
+package org.carlspring.strongbox.cron.jobs;
+
+import org.carlspring.strongbox.cron.domain.CronTaskConfigurationDto;
+
+import java.util.Collection;
+import java.util.Objects;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class PerRepositoryDuplicationStrategy
+        implements CronJobDuplicationStrategy
+{
+
+    private static final PerRepositoryDuplicationStrategy DEFAULT = new PerRepositoryDuplicationStrategy();
+
+    private final String propertyStorageId;
+
+    private final String propertyRepositoryId;
+
+    private PerRepositoryDuplicationStrategy()
+    {
+        this("storageId", "repositoryId");
+    }
+
+    public PerRepositoryDuplicationStrategy(final String propertyStorageId,
+                                            final String propertyRepositoryId)
+    {
+        this.propertyStorageId = propertyStorageId;
+        this.propertyRepositoryId = propertyRepositoryId;
+    }
+
+    public static PerRepositoryDuplicationStrategy getDefault()
+    {
+        return DEFAULT;
+    }
+
+    @Override
+    public boolean duplicates(final CronTaskConfigurationDto candidate,
+                              final Collection<CronTaskConfigurationDto> existing)
+    {
+        if (CollectionUtils.isEmpty(existing))
+        {
+            return false;
+        }
+        return existing.stream().filter(e -> duplicates(candidate, e)).findFirst().isPresent();
+    }
+
+    private boolean duplicates(final CronTaskConfigurationDto first,
+                               final CronTaskConfigurationDto second)
+    {
+        if (first == null || second == null)
+        {
+            return false;
+        }
+
+        final Class<?> firstJobClass = first.getJobClass();
+        final Class<?> secondJobClass = second.getJobClass();
+
+        if (!Objects.equals(firstJobClass, secondJobClass))
```

Comment:
same classes can be not equals if they loaded with different class loaders for example.

Reply:
yes, you're right, thanks